src/engine/strat_engine/liminal.rs

```diff
@@ -316,35 +312,308 @@ pub fn get_blockdevs(
     Ok((datadevs, cachedevs))
 }
 
+/// Info for a discovered Luks Device belonging to Stratis.
+#[derive(Debug, Eq, Hash, PartialEq)]
+pub struct LLuksInfo {
+    /// Generic information + Stratis identifiers
+    pub ids: StratisInfo,
+    pub key_description: KeyDescription,
+}
+
+impl fmt::Display for LLuksInfo {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "{}, key description: {}",
+            self.ids,
+            self.key_description.as_application_str()
+        )
+    }
+}
+
+impl From<LuksInfo> for LLuksInfo {
+    fn from(info: LuksInfo) -> LLuksInfo {
+        LLuksInfo {
+            ids: info.info,
+            key_description: info.key_description,
+        }
+    }
+}
+
+impl<'a> Into<Value> for &'a LLuksInfo {
+    // Precondition: (&StratisInfo).into() pattern matches Value::Object()
+    fn into(self) -> Value {
+        let mut json = json!({
+            "key_description": Value::from(self.key_description.to_system_string())
```

Comment:
Is the reason you've chosen `to_system_string()` for debugging purposes?

Reply:
> Is the reason you've chosen `to_system_string()` for debugging purposes?

I was of two minds about this. I'll change it to ```as_application_str```, at present it doesn't really make any difference to me.